src/mlpack/methods/logistic_regression/logistic_regression_main.cpp

```diff
@@ -141,9 +141,15 @@ static void mlpackMain()
   const double tolerance = CLI::GetParam<double>("tolerance");
   const double stepSize = CLI::GetParam<double>("step_size");
   const size_t batchSize = (size_t) CLI::GetParam<int>("batch_size");
+
+  // Checking that max iterations is non negative
+  if(CLI::GetParam<int>("max_iterations") <= 0) {
+    Log::Fatal << "Max Iterations (" << CLI::GetParam<int>("max_iterations") << ") cannot be negative" << endl;
+  }
```

Comment:
Can you use the `RequireParamValue()` function here please?  I suppose also we should check that the batch size is greater than 0 too.  Also, 0 is an allowable value for the maximum number of iterations.

Reply:
Done.